cmd/lightclient/sentinel/handlers.go

```diff
@@ -14,86 +14,106 @@
+	"github.com/ledgerwatch/erigon/cmd/lightclient/sentinel/proto/p2p"
+	"github.com/ledgerwatch/erigon/cmd/lightclient/sentinel/proto/ssz_snappy"
+
 	"github.com/ledgerwatch/log/v3"
 	"github.com/libp2p/go-libp2p/core/network"
 	"github.com/libp2p/go-libp2p/core/protocol"
-	ssz "github.com/prysmaticlabs/fastssz"
 )
 
 var ProtocolPrefix = "/eth2/beacon_chain/req"
 
 func getHandlers(s *Sentinel) map[protocol.ID]network.StreamHandler {
 	return map[protocol.ID]network.StreamHandler{
-		protocol.ID(ProtocolPrefix + "/ping/1/ssz_snappy"):                   s.pingHandler,
-		protocol.ID(ProtocolPrefix + "/status/1/ssz_snappy"):                 s.statusHandler,
-		protocol.ID(ProtocolPrefix + "/goodbye/1/ssz_snappy"):                s.goodbyeHandler,
+		protocol.ID(ProtocolPrefix + "/ping/1/ssz_snappy"):                   curryStreamHandler(ssz_snappy.NewStreamCodec, pingHandler),
+		protocol.ID(ProtocolPrefix + "/status/1/ssz_snappy"):                 curryStreamHandler(ssz_snappy.NewStreamCodec, statusHandler),
+		protocol.ID(ProtocolPrefix + "/goodbye/1/ssz_snappy"):                curryStreamHandler(ssz_snappy.NewStreamCodec, s.goodbyeHandler),
+		protocol.ID(ProtocolPrefix + "/metadata/1/ssz_snappy"):               curryStreamHandler(ssz_snappy.NewStreamCodec, metadataHandler),
 		protocol.ID(ProtocolPrefix + "/beacon_blocks_by_range/1/ssz_snappy"): s.blocksByRangeHandler,
 		protocol.ID(ProtocolPrefix + "/beacon_blocks_by_root/1/ssz_snappy"):  s.beaconBlocksByRootHandler,
-		protocol.ID(ProtocolPrefix + "/metadata/1/ssz_snappy"):               s.metadataHandler,
 	}
 }
 
-func (s *Sentinel) pingHandler(stream network.Stream) {
-	pingBytes := make([]byte, 8)
-	n, err := stream.Read(pingBytes)
-	if err != nil {
-		log.Warn("handler crashed", "err", err)
-		return
-	}
-	if n != 8 {
-		log.Warn("Invalid ping received")
-		return
-	}
-	log.Info("[Lightclient] Received", "ping", ssz.UnmarshallUint64(pingBytes))
-	// Send it back
-	n, err = stream.Write(pingBytes)
-	if err != nil {
-		log.Warn("handler crashed", "err", err)
-		return
+func setDeadLines(stream network.Stream) {
+	if err := stream.SetReadDeadline(time.Now().Add(clparams.TtfbTimeout)); err != nil {
+		log.Error("failed to set stream read dead line", "err", err)
 	}
-	if n != 8 {
-		log.Warn("Could not send Ping")
+	if err := stream.SetWriteDeadline(time.Now().Add(clparams.RespTimeout)); err != nil {
+		log.Error("failed to set stream write dead line", "err", err)
 	}
 }
 
-func (s *Sentinel) statusHandler(stream network.Stream) {
-
-	log.Info("Got status request")
+// curryStreamHandler converts a func(ctx *proto.StreamContext, dat proto.Packet) error to func(network.Stream)
+// this allows us to write encoding non specific type safe handler without performance overhead
+func curryStreamHandler[T proto.Packet](newcodec func(network.Stream) proto.StreamCodec, fn func(ctx *proto.StreamContext, v T) error) func(network.Stream) {
+	return func(s network.Stream) {
+		setDeadLines(s)
```

Comment:
can you delete this it is not necessary

Reply:
I think he is referring to `setDeadLines` I personally do not see the need to set deadlines on request